karelics_vesc_can_driver/vesc_can_driver.py

```diff
@@ -189,12 +193,20 @@ def can_cb(self, msg: Frame):
             self.get_logger().info(str(e))
 
         # Publish the current status of the vescs in to ros world
-        # TODO Why do we do this for all the vescs? Should we process tick only for the current_vesc?
+        # TODO: Why do we do this for all the vescs? Should we process tick only for the current_vesc?
+        # TODO: talk with mart about this? is it for timestamping and synchronization across vescs?
+        vesc = None
         for vesc in self.known_vescs:
             vesc.tick()
 
+        if vesc is not None:
+            self.publish_battery_state(vesc.v_in)
```

Comment:
Why do you not just add the battery status to each vesc status? Now you only use the last of the know vescs although (vesc from the loop).

Reply:
I would call this in the `vesc.tick` method. not here.

Reply:
This is something I've implemented. Agree, getting the value only from the last vesc doesn't look good, we'll change that. But there are couple reasons why it was done like this and in this place:

- We would like to publish the overall battery voltage status to /battery topic
- All the vescs publish constantly slightly different v_in value, which is why publishing it in .tick() for each vesc would make the overall battery voltage jump up and down a bit. This is why I chose just a single vesc which status we follow.

What do you think if we changed this so that we sum up here the v_in from all the vescs, take the average of that and publish to battery state?

Reply:
I understand why you are doing this. But I would recommend to think about separating concerns.  The way I see it, the vesc driver is responsible for communicating with the vescs period. 

What you are doing now is adding logic about the battery to the node that talks to the motor drivers. I would rather make a battery monitoring node that has knowledge about the battery and takes all sensor values of the vesc in to account. 

Maybe we could redesign the VESC interface to use a standard message linke: http://docs.ros.org/en/api/sensor_msgs/html/msg/BatteryState.html

So the battery node could be independed of the VESC interface

Reply:
Ok, I agree. We'll make this change.